packages/next/src/server/lib/experimental/create-env-definitions.ts

```diff
@@ -0,0 +1,40 @@
+import type { Env } from '@next/env'
+import { join } from 'node:path'
+import { writeFile } from 'node:fs/promises'
+
+export async function createEnvDefinitions({
+  distDir,
+  env,
+  isTest = false,
+}: {
+  distDir: string
+  env: Env
+  isTest?: boolean
+}) {
+  const envKeysStr = Object.keys(env)
+    .map((key) => `      ${key}: readonly string`)
+    .join('\n')
+
+  const definitionStr = `// Type definitions for Next.js environment variables
+declare global {
+  namespace NodeJS {
+    interface ProcessEnv {
+${envKeysStr}
+    }
+  }
+}
+export {}`
+
+  if (isTest) {
+    return definitionStr
+  }
+
+  try {
+    // we expect the types directory to already exist
+    const envDtsPath = join(distDir, 'types', 'env.d.ts')
```

Comment:
Can we check if types dir is created here so we don't have to do that logic for turbopack specifically